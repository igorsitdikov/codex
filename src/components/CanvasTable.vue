<template>
  <table>
    <tr v-for="(row,r_index) in todo" :key="r_index">
      <td v-for="(cell,c_index) in row" :key="c_index" @click="getCell(r_index,c_index)">{{cell}}</td>
    </tr>
  </table>
</template>


<script>
export default {
  name: "CanvasTable",
  props: {
    todo: {
      type: Array,
      required: true
    }
  },
//   data: function(){
//       return {cell : this.point}
//   },
  methods: {
    getCell(x,y) {
        this.$emit('clicked', [x,y]);
    }
  }
};
</script>


<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
td {
  width: 100px;
  height: 108px;
  padding: 0;
  border: 1px solid black;
  text-align: center;
  font-size: 100px;
}

table {
  border-collapse: collapse;
  border: 1px solid black;
}
</style>