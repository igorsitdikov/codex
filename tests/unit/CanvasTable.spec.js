import {createLocalVue, mount, shallowMount} from '@vue/test-utils'
import CanvasTable from '@/components/CanvasTable.vue'

describe("CanvasTable", () => {
    test('is a CanvasTable instance', () => {
        const expected = [
            ["o", "o", "x"],
            ["x", "o", "x"],
            ["x", "o", "x"]
        ];
        const wrapper = mount(CanvasTable, {
            propsData: {
                canvas: expected
            }
        });
        expect(wrapper.isVueInstance()).toBeTruthy()
    });

    test('matches snapshot', () => {
        const expected = [
            ["o", "o", "x"],
            ["x", "o", "x"],
            ["x", "o", "x"]
        ];
        const wrapper = shallowMount(CanvasTable, {
            propsData: {
                canvas: expected
            }
        });
        expect(wrapper.html()).toMatchSnapshot()
    });

    test('matches snapshot', () => {
        const expected = [
            ["o", "o", "o"],
            ["o", "o", "o"],
            ["o", "o", "o"]
        ];
        const wrapper = shallowMount(CanvasTable, {
            propsData: {
                canvas: expected
            }
        });
        expect(wrapper.html()).toMatchSnapshot()
    });
});