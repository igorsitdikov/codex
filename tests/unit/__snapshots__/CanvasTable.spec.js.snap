// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CanvasTable matches snapshot 1`] = `
<table>
  <tr>
    <td>o</td>
    <td>o</td>
    <td>x</td>
  </tr>
  <tr>
    <td>x</td>
    <td>o</td>
    <td>x</td>
  </tr>
  <tr>
    <td>x</td>
    <td>o</td>
    <td>x</td>
  </tr>
</table>
`;

exports[`CanvasTable matches snapshot 2`] = `
<table>
  <tr>
    <td>o</td>
    <td>o</td>
    <td>o</td>
  </tr>
  <tr>
    <td>o</td>
    <td>o</td>
    <td>o</td>
  </tr>
  <tr>
    <td>o</td>
    <td>o</td>
    <td>o</td>
  </tr>
</table>
`;
